src/components/vm/deleteDialog.jsx

```diff
@@ -131,8 +131,12 @@ export class DeleteDialog extends React.Component {
         const storagePools = getVmStoragePools(vm);
 
         Promise.all(
-            (Array.isArray(vm.snapshots) ? vm.snapshots : [])
-                    .map(snapshot => snapshotDelete({ connectionName: vm.connectionName, domainPath: vm.id, snapshotName: snapshot.name }))
+            (Array.isArray(vm.snapshots) ? vm.snapshots : []).map(snapshot => {
+                return snapshotDelete({ connectionName: vm.connectionName, domainPath: vm.id, snapshotName: snapshot.name })
+                        .catch(ex => {
+                            return Promise.reject(new Error(cockpit.format(_("Could not delete snapshot $0: "), snapshot.name) + ex.message));
```

Comment:
There is not need to change this here. If you leave it how it was before the error will be now propagated to the new catch you added in line 160.

After all you probably don't want a message like:

Coult not delete my-vm-name: Could not deleted snapshot vm-vm-snapshot0: error from libvirt 

It's fine to omit the middle string.

Reply:
If we omit that, then there is no mention of the snapshot in error message. And I don't if user would not be confused where exactly does the message come

Reply:
The snapshot APIs error seem to contain the name. In your case it fails on deleting the storage, and the error is again correct.

![Screen Shot 2023-04-14 at 19 25 51](https://user-images.githubusercontent.com/14921356/232114791-250e8312-c989-40ce-954c-c0574a423ac7.png)